addons/website/static/src/scss/website.scss

```diff
@@ -1271,6 +1286,68 @@ header {
     }
 }
 
+// Offcanvas mega menu on mobile or in sidebar/hamburger header
+.o_mega_menu_is_offcanvas {
+    .o_mega_nav, .o_mega_menu {
+        @include transition(opacity $offcanvas-transition-duration ease-in-out, visibility $offcanvas-transition-duration ease-in-out);
+        z-index: $o-mega-menu-zindex;
+        max-width: $offcanvas-horizontal-width;
+        opacity: 0;
+
+        section {
+            @include transition(transform $offcanvas-transition-duration ease-in-out, visibility $offcanvas-transition-duration ease-in-out);
+            transform: translateX(5%);
+        }
+    }
+
+    .o_mega_menu.show, .o_mega_menu.show ~ .o_mega_nav {
+        @include transition(opacity $offcanvas-transition-duration ease-in-out, visibility $offcanvas-transition-duration ease-in-out);
+        visibility: visible !important;
+        opacity: 1;
+
+        section {
+            @include transition(transform $offcanvas-transition-duration ease-in-out, visibility $offcanvas-transition-duration ease-in-out);
+            transform: translateX(0);
+        }
+    }
+
+    .o_mega_nav {
+        padding-left: $grid-gutter-width * .5;
+        height: $o-mega-menu-nav-height;
+    }
+
+    // Div needed for selector specificity to apply mh
+    div.o_mega_menu {
+        position: fixed !important;
+        inset: 0 0 0 auto !important;
+        display: block;
+        visibility: hidden;
+        margin-top: $o-mega-menu-nav-height !important;
+        border: none;
+        max-height: 100vh;
+        background-color: o-color('menu-custom') or o-color('menu');
+        overflow-x: hidden;
+
+        .container > .row {
+            flex-direction: column;
+        }
```

Comment:
Does that change anything? In any case, `.container` is probably wrong as the user can change that.

Reply:
The goal was to make the columns inside the mega_menu take a similar behavior as a "col-12" (full-width column) when it's inside an offcanvas. How is the user able to change the container class? When applying a grid layout in the web_editor it seemed to work fine.

Another approach would be to target the elements with the col classes and setting a flex-basis: auto; (again only in offcanvas rendered mega_menu)